app/main/posts/posts-routes.js

```diff
@@ -1,155 +1,158 @@
-                        break;
-                    default:
-                        var view = $transition$.params().view ? $transition$.params().view : 'map';
-                        if (view === 'list') {
-                            $state.go('list', {view: 'data'}, {reload: true});
-                        }
+            url: '^/collections/:collectionId',
+            name: 'list.collectionRedirector',
+            onEnter: function ($state, $transition$, PostFilters, collection) {
+                if (collection.view === 'data' || collection.view === 'list') {
+                    $state.go('list.data.collection', {collectionId: collection.id, view: 'data'});
+                } else {
+                    $state.go('list.map.collection', {collectionId: collection.id, view: 'map'});
                 }
-
             }
         }
     )
     .state(
         {
-            url: '^/collections/:id/:view',
-            name: 'collection',
-            parent: 'list.data',
+            url: '/views/data',
+            name: 'list.data',
             params: {
-                id: null,
-                view: {squash: true, value: null}
+                view: {value: 'data', squash: true},
+                filterState: {value: null, squash: true},
+                savedSearchState: {value: null, squash: true}
             },
-            controller: require('./collections/collections-controller.js'),
-            template: require('./collections/collections.html'),
-            onEnter: function ($state, $transition$, collection) {
-                var viewParam = collection.view;
-                if (viewParam === 'list') {
-                    viewParam = 'data';
-                } else if (!viewParam) {
-                    viewParam = 'map';
-                } else if ($transition$.params().view && $transition$.params().view !== 'list') {
-                    viewParam = $transition$.params().view;
+            views: {
+                listView: {
+                    controller: require('./views/post-view-data.controller.js'),
+                    template: require('./views/post-view-data.html')
                 }
-                $state.go('collection', {view: viewParam, id: $transition$.params().id});
             },
             resolve: {
-                collection: ['$transition$', 'CollectionEndpoint', function ($transition$, CollectionEndpoint) {
-                    return CollectionEndpoint.get({collectionId: $transition$.params().id}).$promise;
+                //change to selectedPost and refactor the selectedposts in general
+                selectedPost: ['$transition$', 'PostEndpoint', function ($transition$, PostEndpoint) {
+                    if ($transition$.params().postId) {
+                        return PostEndpoint.get({ id: $transition$.params().postId }).$promise;
+                    }
```

Comment:
return null if no postId is available (for consistency)

Reply:
What happens if there's no postId? In core ngRouter a failing resolve condition would cancel the routing. Does the same happen for uirouter?

Reply:
ok so in this scenario we  don't want the resolve to fail if there is no post id, because you are dealing with a possible param in a child state that can be null without the transition needing to fail. Does that make sense? 
If there's no postId, there's nothing wrong with it, it just means yout  don't have a selectedPost.